### PR TITLE
feat: #396 added option to open Image in full screen

### PR DESCRIPTION
**Pull request by FriendOfEveryone**
disabled husky pre-commit 
feat: #396 added option to open Image in full screen

### CODE REVIEW

File: mw-webapp/pnpm-lock.yaml
Comment (Ekaterina1994) [inline mw-webapp/pnpm-lock.yaml:1]: What and why has changed in this file?

File: mw-webapp/src/component/image/Image.module.scss [hunk 1]
```diff
@@ -1,3 +1,15 @@
+.imageWrapperFullScreen {
+  position: fixed;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  width: 100vw;
+  height: 100vh;
+  padding: 7%;
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/image/Image.module.scss:9]: don't use %

File: mw-webapp/src/component/image/Image.tsx [hunk 2]
```diff
@@ -25,21 +26,41 @@ interface ImageProps {
...
   return (
-    <img
-      src={props.src}
-      alt={props.alt}
-      className={className}
-      data-cy={props.dataCy}
-    />
+    <div className={isOpen ? styles.imageWrapperFullScreen : ""}>
+      <img
+        tabIndex={0}
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/image/Image.tsx:55]: why do we need this property?

File: mw-webapp/src/component/image/Image.module.scss [hunk 3]
```diff
@@ -1,3 +1,15 @@
+.imageWrapperFullScreen {
+  position: fixed;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/image/Image.module.scss:6]: Don't enough only top and left for position image?

File: mw-webapp/src/component/image/Image.module.scss [hunk 4]
```diff
@@ -1,3 +1,15 @@
...
+  width: 100vw;
+  height: 100vh;
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/image/Image.module.scss:8]: Isn't is too big image?

File: mw-webapp/src/component/image/Image.tsx [hunk 5]
```diff
@@ -25,21 +26,41 @@ interface ImageProps {
    * Data attribute for cypress testing
    */
   dataCy?: string;
+
+  /**
+   * Should image open?
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/image/Image.tsx:31]: Delete "?" at the end

Add default value

File: mw-webapp/src/component/image/Image.tsx [hunk 6]
```diff
@@ -25,21 +26,41 @@ interface ImageProps {
...
  * Сomponent for displaying images
  */
 export const Image = (props: ImageProps) => {
+
+  const [isOpen, setIsOpen] = useState(false);
+
   const className = clsx(styles.image, props.className);
 
+  /**
+   * Function full screen image
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/image/Image.tsx:46]: Not clear comment

File: mw-webapp/src/component/image/Image.tsx [hunk 7]
```diff
@@ -25,21 +26,41 @@ interface ImageProps {
...
+   */
+  const onClickHandler = (value: boolean) => {
+    setIsOpen(value);
+  };
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/image/Image.tsx:50]: Maybe better don't use additional wrapper and use setIsOpen inside code instead of function-wrapper? What do you think?

File: mw-webapp/src/component/image/Image.tsx [hunk 8]
```diff
@@ -25,21 +26,41 @@ interface ImageProps {
...
+
   return (
-    <img
-      src={props.src}
-      alt={props.alt}
-      className={className}
-      data-cy={props.dataCy}
-    />
+    <div className={isOpen ? styles.imageWrapperFullScreen : ""}>
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/image/Image.tsx:53]:
```suggestion
    <div className={isOpen ?? styles.imageWrapperFullScreen}>
```

File: mw-webapp/src/component/image/Image.tsx [hunk 9]
```diff
@@ -25,21 +26,41 @@ interface ImageProps {
...
+      <img
+        tabIndex={0}
+        src={props.src}
+        alt={props.alt}
+        className={className}
+        data-cy={props.dataCy}
+        onClick={props.isShouldOpen ? () => onClickHandler(true) : undefined}
+        onBlur={() => onClickHandler(false)}
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/image/Image.tsx:61]: Maybe better to use Modal and render big image after click by trigger. 

@tritonJS826 What do you think?

If not, then need add possibility to close image not only by onBlur - it can be not clear for user

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/image/Image.tsx (hunk 2 above)
Comment (FriendOfEveryone) [reply mw-webapp/src/component/image/Image.tsx:55]: Which? tabIndex?

File: mw-webapp/src/component/image/Image.module.scss (hunk 4 above)
Comment (FriendOfEveryone) [reply mw-webapp/src/component/image/Image.module.scss:8]: depending on the orientation of the picture this will change

File: mw-webapp/src/component/image/Image.tsx (hunk 5 above)
Comment (FriendOfEveryone) [reply mw-webapp/src/component/image/Image.tsx:31]: Ok

File: mw-webapp/src/component/image/Image.tsx (hunk 7 above)
Comment (FriendOfEveryone) [reply mw-webapp/src/component/image/Image.tsx:50]: Ok. I will change

File: mw-webapp/src/component/image/Image.tsx (hunk 8 above)
Comment (FriendOfEveryone) [reply mw-webapp/src/component/image/Image.tsx:53]: ok

File: mw-webapp/src/component/image/Image.tsx (hunk 9 above)
Comment (FriendOfEveryone) [reply mw-webapp/src/component/image/Image.tsx:61]: I can add a transparent button in the corner with X

File: mw-webapp/pnpm-lock.yaml
Comment (FriendOfEveryone) [reply mw-webapp/pnpm-lock.yaml:1]: I didn't change anything manually. Reinstalled packages and modules

File: mw-webapp/src/component/image/Image.module.scss (hunk 1 above)
Comment (FriendOfEveryone) [reply mw-webapp/src/component/image/Image.module.scss:9]: Ok

File: mw-webapp/pnpm-lock.yaml
Comment (Ekaterina1994) [reply mw-webapp/pnpm-lock.yaml:1]: Then these changes need to delete

File: mw-webapp/src/component/image/Image.module.scss (hunk 1 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/image/Image.module.scss:9]: Pease in comments write questions , suggestions, or status : done, deleted etc. Ok is not clear

File: mw-webapp/src/component/image/Image.tsx (hunk 2 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/image/Image.tsx:55]: yes

File: mw-webapp/src/component/image/Image.tsx (hunk 8 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/image/Image.tsx:46]: Yes? sory, I was mean
isOpen && styles.imageWrapperFullScreen

Comment (FriendOfEveryone) [reply mw-webapp/src/component/image/Image.tsx:46]: @Ekaterina1994 if "isOpen" will be false then className={false}

Comment (Ekaterina1994) [reply mw-webapp/src/component/image/Image.tsx:46]: The provided code snippet uses a conditional statement to apply a class based on the value of isOpen. Here's how it works:

className: This is a JavaScript property used to set the class names for an HTML element.
{isOpen && styles.imageWrapperFullScreen}: This is a JavaScript expression within curly braces.
isOpen: This is a boolean variable.
&&: This is the logical AND operator.
styles.imageWrapperFullScreen: This assumes styles is an object containing CSS class names. Here, it accesses the imageWrapperFullScreen class from the styles object.
If isOpen is true:

The entire expression evaluates to true because true && anything is always true.
Therefore, the class name applied will be styles.imageWrapperFullScreen.
If isOpen is false:

The expression isOpen && styles.imageWrapperFullScreen evaluates to false because false && anything is always false.
Since the entire expression evaluates to false, no additional class is applied to the element's className property. The element will only have the base classes defined outside the curly braces.
In summary:

When isOpen is false, the element won't have the styles.imageWrapperFullScreen class applied. It will retain any other classes defined for it outside the conditional statement.

Comment (FriendOfEveryone) [reply mw-webapp/src/component/image/Image.tsx:46]: let b = 0 && 1

Comment (FriendOfEveryone) [reply mw-webapp/src/component/image/Image.tsx:46]: @Ekaterina1994 let b = 0 && 1

Comment (FriendOfEveryone) [reply mw-webapp/src/component/image/Image.tsx:46]: @Ekaterina1994 The && operator returns the last argument if everything is true or the first value is false. In this case.
For example let b = 0 && 1. b will be 0

File: mw-webapp/src/component/image/Image.tsx (hunk 2 above)
Comment (FriendOfEveryone) [reply mw-webapp/src/component/image/Image.tsx:54]: @Ekaterina1994 Without it onBlur does not work

File: mw-webapp/src/component/image/Image.module.scss (hunk 4 above)
Comment (FriendOfEveryone) [reply mw-webapp/src/component/image/Image.module.scss:7]: @Ekaterina1994 it is for image wrapper. Image will scale

File: mw-webapp/src/component/image/Image.module.scss (hunk 3 above)
Comment (FriendOfEveryone) [reply mw-webapp/src/component/image/Image.module.scss:4]: @Ekaterina1994 changed

File: mw-webapp/pnpm-lock.yaml
Comment (FriendOfEveryone) [reply mw-webapp/pnpm-lock.yaml:1]: @Ekaterina1994 deleted

File: mw-webapp/src/component/image/Image.module.scss (hunk 1 above)
Comment (FriendOfEveryone) [reply mw-webapp/src/component/image/Image.module.scss]: @Ekaterina1994 changed

File: mw-webapp/src/component/image/Image.tsx (hunk 6 above)
Comment (FriendOfEveryone) [reply mw-webapp/src/component/image/Image.tsx]: @Ekaterina1994 changed

File: mw-webapp/src/component/image/Image.tsx (hunk 7 above)
Comment (FriendOfEveryone) [reply mw-webapp/src/component/image/Image.tsx]: @Ekaterina1994 changed

File: mw-webapp/src/component/image/Image.tsx (hunk 9 above)
Comment (FriendOfEveryone) [reply mw-webapp/src/component/image/Image.tsx]: @Ekaterina1994 added close button

File: mw-webapp/src/component/image/Image.tsx (hunk 2 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/image/Image.tsx:54]: According to the link https://htmlbook.ru/html/a/tabindex

Атрибут tabindex определяет последовательность перехода между ссылками при нажатии на кнопку Tab.

Do we need it?

File: mw-webapp/.env.example
Comment (Ekaterina1994) [inline mw-webapp/.env.example:1]: why did you deletes this file

File: mw-webapp/src/component/image/Image.module.scss [hunk 10]
```diff
@@ -1,3 +1,32 @@
...
+  align-items: center;
+  justify-content: center;
+  backdrop-filter: blur(5px);
+}
+
+.closeImageButton {
+  position: fixed;
+  z-index: 99;
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/image/Image.module.scss:15]: global

File: mw-webapp/src/component/image/Image.module.scss [hunk 11]
```diff
@@ -1,3 +1,32 @@
...
+  top: 30px;
+  width: 70px;
+  height: 70px;
+  border-radius: 50px;
+  margin: 0 auto;
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/image/Image.module.scss:20]: variables

File: mw-webapp/src/component/image/Image.module.scss [hunk 12]
```diff
@@ -1,3 +1,32 @@
...
+  backdrop-filter: blur(5px);
+}
+
+.closeImageButton {
+  position: fixed;
+  z-index: 99;
+  top: 30px;
+  width: 70px;
+  height: 70px;
+  border-radius: 50px;
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/image/Image.module.scss:19]: We don't have such border radius by design

File: mw-webapp/src/component/image/Image.module.scss [hunk 13]
```diff
@@ -1,3 +1,32 @@
...
+  margin: 0 auto;
+  background-color: #aaaaaa;
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/image/Image.module.scss:21]: use global var

File: mw-webapp/src/component/image/Image.tsx [hunk 14]
```diff
@@ -25,21 +26,41 @@ interface ImageProps {
...
+      >
+        CLOSE
+      </button>}
+      <img
+        tabIndex={0}
+        src={props.src}
+        alt={props.alt}
+        className={className}
+        data-cy={props.dataCy}
+        onClick={props.isShouldOpen ? () => setIsOpen(true) : undefined}
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/image/Image.tsx:59]:
```suggestion
        onClick={() => props.isShouldOpen && setIsOpen(true)}
```
